unittests/CMakeLists.txt

```diff
@@ -44,11 +44,25 @@ set(CUDAQ_RUNTIME_TEST_SOURCES
   common/NoiseModelTester.cpp
   integration/tracer_tester.cpp
   integration/gate_library_tester.cpp
+  dynamics/scalar_ops_simple.cpp
+  dynamics/scalar_ops_arithmetic.cpp
+  dynamics/elementary_ops_simple.cpp
+  dynamics/elementary_ops_arithmetic.cpp
+  dynamics/product_operators_arithmetic.cpp
+  dynamics/test_runge_kutta_time_stepper.cpp
+  dynamics/test_runge_kutta_integrator.cpp
+  dynamics/test_helpers.cpp
+  dynamics/rydberg_hamiltonian.cpp
+  dynamics/test_cudm_helpers.cpp
+  dynamics/test_cudm_state.cpp
+  dynamics/test_cudm_time_stepper.cpp
 )
 
 # Make it so we can get function symbols
 set (CMAKE_ENABLE_EXPORTS TRUE)
 
+include_directories(/usr/local/cuda/targets/x86_64-linux/include)
```

Comment:
The CUDA `include` and `lib` paths should be detected using CMake. We probably want to wrap it in a `CUDA_FOUND` block as well since not all build environment has CUDA.